File: app/src/main/java/com/example/pam_project/repositories/tasks/TaskRepository.java
package com.example.pam_project.repositories.tasks;

import com.example.pam_project.features.tasks.list.TaskInformation;
import com.example.pam_project.utils.constants.TaskStatus;

import io.reactivex.Completable;
import io.reactivex.Flowable;

public interface TaskRepository {

    TaskInformation getTask(final long id);

    Completable insertTask(final String name, final String description, final boolean priority, final TaskStatus status, final long listId);

    Completable updateTask(final long id, final String name, final String description, final boolean priority, final TaskStatus status,
                           final long listId);

    Completable updateTask(final long id, final String name, final String description, final boolean priority);

    Completable deleteTask(final long id);
}

File: app/src/main/java/com/example/pam_project/repositories/tasks/TaskRepository.kt
package com.example.pam_project.repositories.tasks

import com.example.pam_project.features.tasks.list.TaskInformation
import com.example.pam_project.utils.constants.TaskStatus
import io.reactivex.Completable
import io.reactivex.Flowable

interface TaskRepository {
    fun getTask(id: Long): Flowable<TaskInformation?>
    fun insertTask(name: String?, description: String?, priority: Boolean, status: TaskStatus?, listId: Long): Completable
    fun updateTask(id: Long, name: String?, description: String?, priority: Boolean, status: TaskStatus?,
                   listId: Long): Completable

    fun updateTask(id: Long, name: String?, description: String?, priority: Boolean): Completable
    fun deleteTask(id: Long): Completable
}